tests/apps/test_graph_dense.py

```diff
@@ -0,0 +1,129 @@
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Unit tests for strawberryfields.apps.graph.dense
+"""
+# pylint: disable=no-self-use,unused-argument
+import numpy as np
+import networkx as nx
+import pytest
+
+from strawberryfields.apps.graph import dense, sample, resize
+
+pytestmark = pytest.mark.apps
+
+
+samples_subgraphs = np.array(
+    [
+        [1, 2, 3, 4],
+        [0, 1, 2, 3],
+        [0, 1, 2, 3],
+        [0, 1, 2, 5],
+        [0, 1, 2, 3],
+        [0, 1, 4, 5],
+        [0, 1, 4, 5],
+        [0, 1, 3, 4],
+        [0, 1, 2, 3],
+        [0, 1, 4, 5],
+    ]
+)
+
+
+@pytest.mark.parametrize("dim", [5])
+class TestFindDense:
+    """Tests for the function ``dense.find_dense``"""
+
+    def test_callable_input(self, adj):
+        """Tests if function returns the correct output given a custom method set by the user"""
+        objective_return = (0, [0])
+
+        def custom_method(*args, **kwargs):
+            """Mockup of custom-method function fed to ``find_dense``"""
+            return objective_return
+
+        result = dense.find_dense(
+            graph=adj,
+            nodes=3,
+            iterations=10,
+            options={"heuristic": {"method": custom_method}},
+        )
+
+        assert result == objective_return
+
+    @pytest.mark.parametrize("methods", dense.METHOD_DICT)
+    def test_valid_input(self, graph, monkeypatch, methods):
+        """Tests if function returns the correct output under normal conditions. The method is here
+        monkey patched to return a known result."""
+        objective_return = (0, [0])
+
+        def custom_method(*args, **kwargs):
+            """Mockup of custom-method function fed to ``find_dense``"""
+            return objective_return
+
+        with monkeypatch.context() as m:
+            m.setattr(dense, "METHOD_DICT", {methods: custom_method})
+
+            result = dense.find_dense(
+                graph=graph,
+                nodes=3,
+                iterations=10,
+                options={"heuristic": {"method": methods}},
+            )
+
+        assert result == objective_return
+
+
+@pytest.mark.parametrize("dim", [5])
+class TestRandomSearch:
+    """Tests for the function ``dense.random_search``"""
+
+    def sampler(self, *args, **kwargs):
+        """Dummy function for sampling subgraphs"""
+
+        return (samples_subgraphs ** 2).tolist()
+
+    def test_densest_outputted(self, graph, monkeypatch):
```

Comment:
Seems like a bit of a strange name, why not 
```suggestion
    def test_returns_densest(self, graph, monkeypatch):
```

Reply:
Thanks @jswinarton !!